ts/sagas/startup.ts

```diff
@@ -132,6 +134,13 @@ function* initializeApplicationSaga(): IterableIterator<Effect> {
   // User is autenticated, session token is valid
   //
 
+  const walletToken: Option<string> = yield select(walletTokenSelector);
```

Comment:
instead of reading it from the store, you already have what you need in `maybeSessionInfo` that is guaranteed to be a `Some<>` (see lines 80-90s)

Reply:
Okay, my only doubt is: `maybeSessionInformation` and `maybeSessionInfo` seem to be storing the same kind of information (the only difference is the source of the information (backend or store)). `maybeSessionInfo` is scoped to only exist within the `if` @87-96, so if it would not exist any longer when the token needs to be fetched.

I would say there are two options here:
1) make `maybeSessionInformation` a variable rather than a constant and update it if the stored value is none. This would require a `no-let` lint exception, but it would imho improve in readability when compared to option 2)

2) 
```typescript
  const sessionInfoFromStore: ReturnType<
    typeof sessionInfoSelector
  > = yield select(sessionInfoSelector);

  const maybeSessionInformation =
    isSessionRefreshed || maybeSessionInformation.isNone()
      ? yield call(loadSessionInformationSaga, backendClient.getSession)
      : sessionInfoFromStore;
  if (maybeSessionInformation.isNone()) {
    // we can't go further without session info, let's restart
    // the initialization process
    yield put(startApplicationInitialization);
    return;
  }
```

I'm going with option 1) for the time being